coderd/database/migrations/000298_provisioner_jobs_status_idx.up.sql

```diff
@@ -0,0 +1 @@
+CREATE INDEX idx_provisioner_jobs_status ON provisioner_jobs USING btree (job_status);
```

Comment:
Review: it took me a minute to figure out if this is supported, but apparently it is! The [docs on generated columns](https://www.postgresql.org/docs/16/ddl-generated-columns.html) specify that only stored generated columns are supported, and these are always written to disk. TIL!

Reply:
@johnstcn 
yes, that's what I figured as well
Also I thought about using `Hash` Index instead of `BTree`, because so far we're only using `=` or `!=` operations, but wasn't sure, maybe we'll need other operations in the future.

Reply:
TBH this is probably fine, almost all of our other indexes are using BTree. If you were curious, I suppose you could validate with some EXPLAINs.